geoopt/manifolds/siegel/bounded_domain.py

```diff
@@ -0,0 +1,189 @@
+from typing import Optional, Tuple, Union
+import torch
+from geoopt import linalg as lalg
+from .siegel import SiegelManifold
+from .upper_half import UpperHalf
+from .vvd_metrics import SiegelMetricType
+from ..siegel import csym_math as sm
+
+__all__ = ["BoundedDomain"]
+
+
+class BoundedDomain(SiegelManifold):
+    r"""
+    Bounded domain Manifold.
+    This model generalizes the Poincare ball model.
+    Points in the space are complex symmetric matrices.
+
+    .. math::
+
+        \mathcal{B}_n := \{ Z \in \operatorname{Sym}(n, \mathbb{C}) | Id - Z^*Z >> 0 \}
+
+    Parameters
+    ----------
+    metric: str
+        one of "riem" (Riemannian), "fone": Finsler One, "finf": Finsler Infinity,
+        "fmin": Finsler metric of minimum entropy, "wsum": learnable weighted sum.
+    rank: int
+        Rank of the space. Only mandatory for "fmin" and "wsum" metrics.
+    """
+
+    name = "Bounded Domain"
+
+    def __init__(self, metric: str = SiegelMetricType.RIEMANNIAN.value, rank: int = None):
```

Comment:
With `Enum` you can do the following

```
from enum import Enum
class SiegelMetricType(Enum):
    """Supported metric types for Siegel Spaces"""
    RIEMANNIAN = "riem"
    FINSLER_ONE = "fone"
    FINSLER_INFINITY = "finf"
    FINSLER_MINIMUM = "fmin"
    WEIGHTED_SUM = "wsum"
```
and later
```SiegelMetricType("riem")
<SiegelMetricType.RIEMANNIAN: 'riem'>
```
or 
```
SiegelMetricType(SiegelMetricType.RIEMANNIAN)
<SiegelMetricType.RIEMANNIAN: 'riem'>
```
so no need in `from_str` method

Reply:
Done, thanks!